### PR TITLE
Minor cleanups.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/multigrid/mg_constrained_dofs.h [hunk 1]
```diff
@@ -311,60 +312,54 @@ MGConstrainedDoFs::initialize(
 
       // Loop through relevant cells and faces finding those which are periodic
       // neighbors.
-      typename DoFHandler<dim, spacedim>::cell_iterator cell = dof.begin(l),
-                                                        endc = dof.end(l);
-      for (; cell != endc; ++cell)
-        if (cell->level_subdomain_id() != numbers::artificial_subdomain_id)
-          {
-            for (auto f : cell->face_indices())
```
Comment (bangerth) [inline include/deal.II/multigrid/mg_constrained_dofs.h:319]: From this line on, only the indentation changed because I removed the curly brace.

**Review by marcfehling — APPROVED** [review]

File: include/deal.II/multigrid/mg_constrained_dofs.h [hunk 2]
```diff
@@ -311,60 +312,54 @@ MGConstrainedDoFs::initialize(
...
-                        cell->periodic_neighbor(f)->level_subdomain_id() !=
-                          numbers::artificial_subdomain_id,
-                        ExcMessage(
-                          "Periodic neighbor of a locally owned cell must either be owned or ghost."));
-                    }
-                  // Cell is a level-ghost and its neighbor is a
-                  // level-artificial cell nothing to do here
-                  else if (cell->periodic_neighbor(f)->level_subdomain_id() ==
-                           numbers::artificial_subdomain_id)
+      for (const auto &cell : dof.active_cell_iterators_on_level())
```
Comment (marcfehling) [inline include/deal.II/multigrid/mg_constrained_dofs.h:315]:
```suggestion
      for (const auto &cell : dof.active_cell_iterators_on_level(l))
```

Comment (bangerth) [reply include/deal.II/multigrid/mg_constrained_dofs.h]: Oh, oops. Of course!

**Review by marcfehling — APPROVED** [review]

File: include/deal.II/multigrid/mg_constrained_dofs.h [hunk 3]
```diff
@@ -311,60 +312,54 @@ MGConstrainedDoFs::initialize(
...
-                        cell->periodic_neighbor(f)->level_subdomain_id() !=
-                          numbers::artificial_subdomain_id,
-                        ExcMessage(
-                          "Periodic neighbor of a locally owned cell must either be owned or ghost."));
-                    }
-                  // Cell is a level-ghost and its neighbor is a
-                  // level-artificial cell nothing to do here
-                  else if (cell->periodic_neighbor(f)->level_subdomain_id() ==
-                           numbers::artificial_subdomain_id)
+      for (const auto &cell : dof.active_cell_iterators_on_level(l))
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_constrained_dofs.h:315]: Shouldn't this be
```suggestion
      for (const auto &cell : dof.cell_iterators_on_level(l))
```
Some tests fail (in fact, I am surprised that not more are failing), and I believe this is the reason.

**Review by kronbichler — APPROVED** [review]

**Comment by bangerth** [comment]
So many mistakes in such a small patch :-(

Regardless, it seems to work now!

**Pull request merged**